sda-download/api/api.md

```diff
@@ -122,7 +122,9 @@ Headers:
 
 - `Authorization: Bearer <token>` 
 - `Range: bytes=<start>-<end>`  exact positions. Overrides parameter coordinates.
-- `User-agent` used in communication with htsget, to mark who is making the request.
+- `Client-public-key: <key>` used for re-encrypting the header of the file before sending it.
+- `Server-public-key: <key>` used in communication with htsget, for re-encrypting the header of the file.
+- `User-Agent` used in communication with htsget, to mark who is making the request. For example `htsget-search/0.6.6.`
```

Comment:
This is utter nonsense, `download` shouldn't care if the request comes from `htsget` or not.
The only thing we actually needs is the public key in the header if all outgoing archive data is sent re-encrypted

ping @dbampalikis for notification.

Reply:
I don't think `utter nonsense` is a very nice review comment. It is the way we have decided to implement the different headers in download for this version. The reason behind that is that we don't know how the clients will act, given that there are no clients available currently. The assumption is that the client will send all headers (including the `client-public-key` and the `authorization`) to htsget, and then htsget will move them to the sda-download, which should have a way to figure out which one is relevant. Although I understand that there might be better solutions (for example sda-download to pick the key based on whether the `server` is set), this is the one picked from the people that participated in the planning this.
As discussed in a previous comment from Pontus, that can be fixed in the future, but imho it is not a showstopper